### DISCUSSION TITLE
pip install -e .[development] failing

### DISCUSSION THREAD

**Discussion by b-goldsmith** [unitaryfund/mitiq#1986] [General]
I was following the instructions on the Contributing page (https://mitiq.readthedocs.io/en/stable/contributing.html) and when I tried to run the command in step 5, "pip install -e .[development]" the command failed with "no matches found: .[development].

(mitiqenv) briangoldsmith@Brians-MBP mitiq % pip install -e .[development]
zsh: no matches found: .[development]

I then ran "grep -r 'development'" and saw the command in the Makefile. I subsequently ran "make install" which ran the "pip install -e .[development] without issue.

(mitiqenv) briangoldsmith@Brians-MBP mitiq % make install
pip install -e .[development]
Obtaining file:///Users/briangoldsmith/Documents/Mitiq/mitiq
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
..... 

The conda environment was just created following the steps provided.
(mitiqenv) briangoldsmith@Brians-MBP mitiq % python --version
Python 3.11.4

This is being run on a a 2021 MacBook Pro with an M1 Pro chip.

**Comment by purva-thakre** [comment]
@bdg221 I think this might be a mac specific error.  Unless you forgot to cd to the mitiq folder after activating your virtual environment. I just tested it out on linux mint with a new environment.

My workflow is below. I don't use conda for virtual environments though which makes it different from what's in the documentation:

```
virtuanenv mitiq_test
source mitiq_test/bin/activate
cd mitiq
git checkout master
pip install -e .[development]
```
Edit: I just figured out there needs to be quotations in this. Can you please give it a try with one of the following? I can't test it out on my device because it works without the quotations on my device. 
```
pip install -e ".[development]"
pip install -e '.[development]'
```
https://stackoverflow.com/questions/30539798/zsh-no-matches-found-requestssecurity

**Reply by b-goldsmith** [reply]
You were 100% correct with the quotes. I just tested it and it went through perfectly. Can I create an issue to add the quotes to the example in the docs. I know this is a super simple fix, but figure it would be good practice.

**Reply by purva-thakre** [reply]
Go for it. 

Although, I would say keep the instructions the way they are for now. ~~And~~ Add a new line below that if you get an error along the lines of  `zsh: no matches found: .[development]`, try the option with quotation marks. 

I don't know why you go this error and maybe someone else who reviews that PR might know more about it.

**Reply by b-goldsmith** [reply]
I was thinking the same thing especially since you didn't hit the issue. 

From your stackoverflow link, it looked it is related to "zsh" for the shell. By default, it does "[globbing](https://en.wikipedia.org/wiki/Glob_%28programming%29)" and so it requires the quotes. I switched to "bash" and did not have to use the quotes.

**Reply by b-goldsmith** [reply]
I created issue: https://github.com/unitaryfund/mitiq/issues/1988

I would love for it to be assigned to me and I would then create a PR and branch for the change.

**Comment by purva-thakre** [comment]
Closing this discussion as an issue has been created.
